flight/odlc_boundaries/search_path.py

```diff
@@ -0,0 +1,205 @@
+"""
+Functions for generating search paths to cover an area for finding the standard odlc objects
+"""
+
+from shapely.geometry import Polygon
+import utm
+from execute import move_to
+from mavsdk import System
+import logging
+import asyncio
+
+
+def latlon_to_utm(coords: dict[str, float | int | str]) -> dict[str, float | int | str]:
+    """
+    Converts latlon coordinates to utm coordinates and adds the data to the dictionary
+
+    Parameters
+    ----------
+    coords : Dict[str, float]
+        A dictionary containing lat long coordinates
+
+    Returns
+    -------
+    coords : Dict[str, float]
+        An updated dictionary with additional keys and values with utm data
+    """
+
+    utm_coords: tuple[float, float, int, str] = utm.from_latlon(
+        coords["latitude"], coords["longitude"]
+    )
+    coords["utm_x"] = utm_coords[0]
+    coords["utm_y"] = utm_coords[1]
+    coords["utm_zone_number"] = utm_coords[2]
+    coords["utm_zone_letter"] = utm_coords[3]
+    return coords
+
+
+def all_latlon_to_utm(
+    list_of_coords: list[dict[str, float | int | str]]
+) -> list[dict[str, float | int | str]]:
+    """
+    Converts a list of dictionaries with latlon data to add utm data
+
+    Parameters
+    ----------
+    list_of_coords : list[dict[str, union[float, int, str]]]
+        A list of dictionaries that contain lat long data
+
+    Returns
+    -------
+    list_of_coords : [dict[str, union[float, int, str]]]
+        An updated list of dictionaries with added utm data
+    """
+    coord: dict[str, float | int | str]
```

Comment:
This guy's actual type is an integer, so its type annotation should be:
coord: int